pandas/core/indexes/base.py

```diff
@@ -1385,6 +1398,12 @@ def set_names(self, names, level=None, inplace: bool = False):
             idx = self
         else:
             idx = self._shallow_copy()
+
+        if isinstance(self, ABCMultiIndex) and is_dict_like(names):
```

Comment:
add a comment on what this condition is doing?

Reply:
Done